### PR TITLE
[JS] Override the `Doc.external`-getter to avoid `alert`-modals on load (issue 15509)

### LINKED ISSUE #15509: Fix warning: Please save this document on your local desktop, to have it function properly
Link to PDF file:
[3c4347e5-f117-48af-9349-43e9f72ea811.pdf](https://github.com/mozilla/pdf.js/files/9643207/3c4347e5-f117-48af-9349-43e9f72ea811.pdf)

Configuration:
- Web browser and its version: 105.0.1 (64-bit)
- Operating system and its version: Windows 10
- Is a browser extension: No

Steps to reproduce the problem:
1. Open the pdf in firebox pdf viewer. A warning will be thrown. **Please save this document on your local desktop, to have it function properly**

What is the expected behavior? 
The pdf should load without any warning. The behavior to edit the pdf should not change.

What went wrong?
image

Link to a viewer: [Pdf Viewer](https://mozilla.github.io/pdf.js/web/viewer.html
)

### PR DESCRIPTION
**Pull request by Snuffleupagus**
This property is documented in https://web.archive.org/web/20201112021418if_/https://www.adobe.com/content/dam/acom/en/devnet/acrobat/pdfs/AcrobatDC_js_api_reference.pdf#G5.1977075

Given that PR #14207, which is *somewhat* similar, landed without tests that's hopefully fine here as well.

### CODE REVIEW

File: src/scripting_api/doc.js [hunk 1]
```diff
@@ -272,7 +274,7 @@ class Doc extends PDFObject {
   }
 
   get external() {
-    return true;
+    return DOC_EXTERNAL;
```
Comment (calixteman) [inline src/scripting_api/doc.js:277]: Maybe you could just add a small explanation on why we aren't "correctly" implementing this getter.

**Review by calixteman — APPROVED** [review]
LGTM.

File: src/scripting_api/doc.js (hunk 1 above)
Comment (Snuffleupagus) [reply src/scripting_api/doc.js:277]: Sure, fixed now!

**Pull request merged**